sklearn/utils/extmath.py

```diff
@@ -715,6 +715,41 @@ def _incremental_mean_and_var(X, last_mean=.0, last_variance=None,
     return updated_mean, updated_variance, updated_sample_count
 
 
+def _incremental_mean_and_var_nan(X, last_mean=.0, last_variance=None,
```

Comment:
I would factorize `_invremental_mean_and_var` and `_incremental_mean_and_var` to avoid code duplication.
I could think of a new argument in `_incremental_mean_and_var` (e.g. `allow_nan=False`) and modify a single line:

```python
if allow_nan:
    var_func = np.nanvar
else:
    var_func = np.var
...
new_unnormalized_variance = var_func(X, axis=0) * new_sample_count
```